packages/idyll-ast/src/index.js

```diff
@@ -53,11 +62,30 @@ const getText = function(node) {
 
 const walkNodes = function(ast, f) {
   (ast || []).forEach((node) => {
-    walkNodes(getChildren(node), f);
+    walkNodes(getChildren(node), f); 
     f(node);
   });
 };
 
+/**
+ * function to do a depth first traversal on the ast tree. 
+ * @param {*} ast     Array that forms the tree structure 
+ * @param {*} f       call-back function 
+ */
+const dfsTraversal = function (ast, f) {
```

Comment:
Since we already have a function named `walkNodes`, maybe `walkNodesDepthFirst` or something would be more  consistent?